servers/dsub/jobs/controllers/aggregation_controller.py

```diff
@@ -24,47 +19,110 @@ def get_job_aggregations(timeFrame, projectId=None):
-            StatusCount(status='Succeeded', count=7),
-            StatusCount(status='Failed', count=4)
-        ]))
-
-    ownerAggregation = Aggregation(
-        name='AnotherLabel', key='job-id', entries=[entry1, entry2])
-
-    projectAggregation = Aggregation(
-        name='User', key='userId', entries=[userEntry1, userEntry2])
+    window_min = time_frame.time_frame_to_start_time(timeFrame)
+    provider = providers.get_provider(_provider_type(), projectId,
+                                      _auth_token())
+
+    jobs = jobs_generator.generate_jobs_by_window(provider, projectId,
+                                                  window_min)
+
+    total_summary = {}
+    user_summary = {}
+    job_name_summary = {}
+    label_summaries = {}
+    label_freq = {}
+
+    for job in jobs:
+        _count_total_summary(job, total_summary)
+        _count_for_key(job, user_summary, lambda j: j.extensions.user_id)
+        _count_for_key(job, job_name_summary, lambda j: j.name)
+        _count_top_labels(label_summaries, job, label_freq)
+
+    aggregations = [
+        _to_aggregation('User Id', 'userId', user_summary),
+        _to_aggregation('Job Name', 'name', job_name_summary)
+    ]
+
+    _to_top_labels_aggregations(label_freq, label_summaries, aggregations)
 
     return AggregationResponse(
-        summary=summaryCounts,
-        aggregations=[ownerAggregation, projectAggregation])
+        summary=_to_summary_counts(total_summary), aggregations=aggregations)
+
+
+def _count_total_summary(job, total_summary):
+    if job.status not in total_summary:
+        total_summary[job.status] = 0
+    total_summary[job.status] += 1
+
+
+def _count_for_key(job, summary, key_lambda):
+    key = key_lambda(job)
+
+    count = summary.get(key, {})
+
+    if job.status not in count:
+        count[job.status] = 0
+    count[job.status] += 1
+
+    summary[key] = count
+
+
+def _count_top_labels(label_summaries, job, label_freq):
+    for label in job.labels:
+        label_freq[label] = label_freq.get(label, 0) + 1
+
+        label_summary = label_summaries.get(label, {})
+        _count_for_key(job, label_summary, lambda j: j.labels[label])
+        label_summaries[label] = label_summary
+
+
+def _to_summary_counts(summary_counts):
+    return StatusCounts([
+        StatusCount(status, count) for status, count in summary_counts.items()
+    ])
+
+
+def _to_aggregation(name, key, summary):
+    entries = []
+
+    for item, counts_dict in summary.items():
+        counts_list = []
+
+        for status, count in counts_dict.items():
+            counts_list.append(StatusCount(status=status, count=count))
+
+        entries.append(
+            AggregationEntry(
+                label=item, status_counts=StatusCounts(counts_list)))
+
+    return Aggregation(name=name, key=key, entries=entries)
+
+
+def _to_top_labels_aggregations(label_freq, label_summaries, aggregations):
+    num_label = min(len(label_freq), _NUM_TOP_LABEL)
```

Comment:
Please leave a `# TODO` to somehow truncate long value lists. For example, if there is a unique label value per job, we wouldn't want to list the value for every job. Eventually it would probably make sense to factor this into which labels we're picking as well.